File: controllers/ItemController.cs
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Backend.Services;
using Backend.Models;
using Microsoft.AspNetCore.Mvc;

namespace Backend.Controllers
{
    [Route("api/[controller]")]
    public class ItemController : ControllerBase
    {
        public ItemController(ItemService service)
        {
            this.service = service;
        }

        readonly ItemService service;

        [HttpGet]
        public async Task<IReadOnlyList<Item>> GetAll()
        {
            return await this.service.GetAll();
        }
    }
}

File: src/main/java/controllers/ItemController.java
package controllers;

import java.util.List;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import models.Item;
import services.ItemService;

@RestController
@RequestMapping("api/item")
public class ItemController {
	public ItemController(ItemService itemService) {
		super();
		this.itemService = itemService;
	}

	private final ItemService itemService;

    @GetMapping
    public List<Item> getAll() {
        return this.itemService.getAll();
    }
}
